### PR TITLE
Multiple bug fixes in crashes

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

File: osquery/tables/system/darwin/crashes.cpp [hunk 1]
```diff
@@ -93,27 +98,31 @@ void readCrashDump(const std::string& app_log, Row& r) {
...
       // If the token is the Crashed thread, update the format string so
       // we can grab the stack trace later.
-      auto t = split(toks[1], " ");
-      if (t.size() == 0) {
+      auto t =
+          toks.size() >= 2 ? split(toks[1], " ") : std::vector<std::string>();
+      if (t.size() < 1) {
+        continue;
+      }
+      unsigned long formatCrashedThread =
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]: So Windows, much sorrow.

```cpp
auto crashed_thread = std::strtoul(t[0].c_str(), nullptr, 10);
```

File: osquery/tables/system/darwin/crashes.cpp [hunk 2]
```diff
@@ -93,27 +98,31 @@ void readCrashDump(const std::string& app_log, Row& r) {
...
+          toks.size() == 4 ? toks[1] + ":" + toks[2] + ":" + toks[3] : "";
     } else if (toks[0] == "Crashed Thread" ||
                toks[0] == "Triggered by Thread") {
       // If the token is the Crashed thread, update the format string so
       // we can grab the stack trace later.
-      auto t = split(toks[1], " ");
-      if (t.size() == 0) {
+      auto t =
+          toks.size() >= 2 ? split(toks[1], " ") : std::vector<std::string>();
+      if (t.size() < 1) {
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]:
```cpp
if (t.empty()) {
  continue;
}
```

File: osquery/tables/system/darwin/crashes.cpp [hunk 3]
```diff
@@ -122,14 +131,22 @@ void readCrashDump(const std::string& app_log, Row& r) {
       boost::smatch results;
       if (boost::regex_search(line, results, e)) {
         auto pid_str = std::string(results[0].first, results[0].second);
-        auto pid = pid_str.substr(1, pid_str.size() - 2);
-        r[kCrashDumpKeys.at(toks[0])] = pid;
+        unsigned long pid = std::strtoul(
+            pid_str.substr(1, pid_str.size() - 2).c_str(), nullptr, 10);
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]: You may still except here if `pid_str.size() <= 2`.

**Comment by facebook-github-bot-0** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2447/files/f9b92bf551c2cf2579c341d79f86680af10e83e5..c2a6236b37e782f3616124b6a111a42101ac6317)

**Pull request merged**